### ISSUE TITLE
508-defect-1 ❗ Launchblocker [COGNITION, AXE-CORE]: Forms with legal, financial, or data functions MUST protect users from errors

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#16242]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Jennifer

## User Story or Problem Statement

As a user with cognitive considerations, I need an interface that anticipates errors and delivers an UI that minimizes the possibility of errors.

## Details

 Forms with legal, financial, or data functions **must** protect users from errors [WCAG 2.0 Level AA - 3.3.4](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-reversible.html). The goal is to minimize the possibility of users making an error that could impact their legal, financial, or data well-being. The checkboxes introduce the possibility of a user getting stuck, not understanding what the issue is with the request to only select two checkboxes, when checkboxes allow for selection of all. The recommendation here is to change the form controls from checkboxes to two select elements, as well as adjust the preceding content to include a `<legend>`, documented in https://github.com/department-of-veterans-affairs/va.gov-team/issues/16261.

## Acceptance Criteria

- [ ] Users are only able to select a maximum of two times

## Environment

* Operating System: all
* Browser: all
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/decision-reviews/higher-level-review/request-higher-level-review-form-20-0996/introduction ` in browser
2. Navigate to "Step 3 of 4: Request an informal conference"
3. Verify that it is possible to select all four checkboxes, when the text says up to two.

### Current code

```diff
<fieldset class="schemaform-field-template">
  <legend class="schemaform-label" id="root_informalConferenceTimes-label">
    <strong>We’ll call <span class="contact-choice selected-rep">your representative</span><span class="contact-choice selected-me">you</span> first to schedule the informal conference. Please choose up to two time periods when it’s best to call <span class="contact-choice selected-rep">your representative</span><span class="contact-choice selected-me">you</span>.</strong>
    <span class="schemaform-required-span">(*Required)</span>
  </legend>
  <div class="schemaform-widget-wrapper">
-   <fieldset>
      <div class="schemaform-field-container schemaform-block">
        <div class="schemaform-block-header"></div>
        <div>
          <div class="schemaform-field-template schemaform-first-field">
            <div class="schemaform-widget-wrapper">
-              <div class="form-checkbox">
-                <input checked id="root_informalConferenceTimes_time0800to1000" name="root_informalConferenceTimes_time0800to1000" type="checkbox">
-                <label class="schemaform-label" for="root_informalConferenceTimes_time0800to1000"><span class="time-title">8:00 a.m. to 10:00 a.m. ET</span></label>
-              </div>
            </div>
-            <div></div>
          </div>
        </div>
        <div>
          <div class="schemaform-field-template">
            <div class="schemaform-widget-wrapper">
-              <div class="form-checkbox">
-                <input id="root_informalConferenceTimes_time1000to1230" name="root_informalConferenceTimes_time1000to1230" type="checkbox">
-                <label class="schemaform-label" for="root_informalConferenceTimes_time1000to1230"><span class="time-title">10:00 a.m. to 12:30 p.m. ET</span></label>
-              </div>
            </div>
-            <div></div>
          </div>
        </div>
        <div>
          <div class="schemaform-field-template">
            <div class="schemaform-widget-wrapper">
-              <div class="form-checkbox">
-                <input id="root_informalConferenceTimes_time1230to1400" name="root_informalConferenceTimes_time1230to1400" type="checkbox">
-                <label class="schemaform-label" for="root_informalConferenceTimes_time1230to1400"><span class="time-title">12:30 p.m. to 2:00 p.m. ET</span></label>
-              </div>
            </div>
-           <div></div>
          </div>
        </div>
        <div>
          <div class="schemaform-field-template">
            <div class="schemaform-widget-wrapper">
-              <div class="form-checkbox">
-                <input id="root_informalConferenceTimes_time1400to1630" name="root_informalConferenceTimes_time1400to1630" type="checkbox">
-                <label class="schemaform-label" for="root_informalConferenceTimes_time1400to1630"><span class="time-title">2:00 p.m. to 4:30 p.m. ET</span></label>
-              </div>
            </div>
-           <div></div>
          </div>
        </div>
      </div>
-   </fieldset>
  </div>
- <div></div>
</fieldset>
```

### Potential solution

```diff
<fieldset class="schemaform-field-template">
  <legend class="schemaform-label" id="root_informalConferenceTimes-label">
    <strong>We’ll call <span class="contact-choice selected-rep">your representative</span><span class="contact-choice selected-me">you</span> first to schedule the informal conference. </strong>
    Please select up to two time periods when it’s best to call <span class="contact-choice selected-rep">your representative</span><span class="contact-choice selected-me">you</span>.
  </legend>
  <div class="schemaform-widget-wrapper">
      <div class="schemaform-field-container schemaform-block">
        <div class="schemaform-block-header"></div>
        <div>
          <div class="schemaform-field-template schemaform-first-field">
            <div class="schemaform-widget-wrapper">
              <label for="options-a1">
                Please select a preferred time. <span class="schemaform-required-span">(*Required)</span>
              </label>
<!-- check with your designers and Peggy to see what the labels should be; an alternate option shown in the recommendation screenshot following -->
              <select class="usa-select" name="options-a1" id="options-a1">
                <option value>- Select -</option>
                <option value="value-a1">8:00 a.m. to 10:00 a.m. ET</option>
                <option value="value-a2" selected>10:00 a.m. to 12:30 p.m. ET</option>
                <option value="value-a3">12:30 p.m. to 2:00 p.m. ET</option>
                <option value="value-a4">2:00 p.m. to 4:30 p.m. ET</option>
              </select>
            </div>
          </div>
        </div>
        <div>
          <div class="schemaform-field-template schemaform-first-field">
            <div class="schemaform-widget-wrapper">
              <label for="options-b1">
                Please select another preferred time.
              </label>
              <select class="usa-select" name="options-b1" id="options-b1">
                <option value>- Select -</option>
                <option value="value-b1">8:00 a.m. to 10:00 a.m. ET</option>
                <!-- <option value="value-b2">10:00 a.m. to 12:30 p.m. ET</option> -->
                <option value="value-b3">12:30 p.m. to 2:00 p.m. ET</option>
                <option value="value-b4">2:00 p.m. to 4:30 p.m. ET</option>
              </select>
            </div>
          </div>
        </div>

      </div>
  </div>
</fieldset>
```

## WCAG or Vendor Guidance (optional)

* [WCAG 2.0 Level AA - 3.3.4](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-reversible.html) - Forms with legal, financial, or data functions MUST protect users from errors.

* [Wuhcag 3.3.4 Level AA - 3.3.4 – Error Prevention (Legal, Financial, Data)](https://www.wuhcag.com/error-prevention-legal-financial-data/) - Reduce the risk of input errors for sensitive data

### Current

![Screen Shot 2020-11-16 at 5 05 06 PM](https://user-images.githubusercontent.com/57469/99315076-11f8ee80-2830-11eb-91e8-1e00264ba44e.png)

### Recommendation

![Screen Shot 2020-11-16 at 5 46 24 PM](https://user-images.githubusercontent.com/57469/99317643-61d9b480-2834-11eb-8b53-d9fbc4a72121.png)

**Comment by peggygannon** [comment]
HI, @Mottie want to confirm that the recommended screenshot is replacing the current one? 

Will these fields have dropdown options from which users can choose?

**Comment by Mottie** [comment]
I have slightly different language in the screenshot I shared in Slack... we've updated the content since this ticket was opened.

**Comment by peggygannon** [comment]
Hi, @Mottie content recommendations for this screen. Please let me know if you think this'll work. 

content for "you"

**We'll call you to schedule an informal conference**
Please provide 1 or 2  preferred times for a call 

Choose the best time for us to call you (*Required)
{time period drop down}

validation error: Please select a time 

Choose another time for us to call you
{time period drop down}


--------

content for "representative"
**We'll call your representative to schedule an informal conference**
Please provide 1 or 2  preferred times for a call

Choose the best time for us to call your representative (*Required)
{time period drop down}

Validation error: Please select a time

Choose another time for us to call your representative
{time period drop down}

**Comment by Mottie** [comment]
Work completed in https://github.com/department-of-veterans-affairs/vets-website/pull/15391.

**Comment by joshkimux** [comment]
I've tested and confirmed that this has been fixed 👍 

<img width="795" alt="Screen Shot 2020-12-28 at 11 44 55 AM" src="https://user-images.githubusercontent.com/14154792/103229933-1c63d900-4902-11eb-8670-cf6b910c5db0.png">
